dom.bs

```diff
@@ -1413,6 +1413,188 @@ can only be used to influence an ongoing one.
+    // But also, watch for signals:
+    signal.addEventListener('abort', () => {
+      // Stop doing amazingness, and:
+      reject(new DOMException('Aborted', 'AbortError'));
+    });
+  });
+}
+</code></pre>
+
+ <p>APIs that require more granular control could extend both {{AbortController}} and
+ {{AbortSignal}} objects according to their needs.
+</div>
+
+
+<h3 id=interface-abortcontroller>Interface {{AbortController}}</h3>
+
+<pre class="idl">
+[Constructor,
+ Exposed=(Window,Worker)]
+interface AbortController {
+  [SameObject] readonly attribute AbortSignal signal;
+
+  void abort();
+};</pre>
+
+<dl class=domintro>
+ <dt><code><var>controller</var> = new <a constructor lt=AbortController()>AbortController</a>()</code>
+ <dd>Returns a new <var>controller</var> whose {{AbortController/signal}} is set to a newly
+ created {{AbortSignal}} object.
+
+ <dt><code><var>controller</var> . <a attribute for=AbortController>signal</a></code>
+ <dd>Returns the {{AbortSignal}} object associated with this object.
+
+ <dt><code><var>controller</var> . <a method for=AbortController lt=abort()>abort</a>()</code>
+ <dd>Invoking this method will set this object's {{AbortSignal}}'s [=AbortSignal/aborted flag=] and
+ signal to any observers that the associated activity is to be aborted.
+</dl>
+
+<p>An {{AbortController}} object has an associated <dfn for=AbortController>signal</dfn> (an
+{{AbortSignal}} object).
+
+<p>The <dfn constructor for=AbortController><code>AbortController()</code></dfn> constructor, when
+invoked, must run these steps:
+
+<ol>
+ <li><p>Let <var>signal</var> be a new {{AbortSignal}} object.
+
+ <li><p>Let <var>controller</var> be a new {{AbortController}} object whose
+ <a for=AbortController>signal</a> is <var>signal</var>.
+
+ <li><p>Return <var>controller</var>.
+</ol>
+
+<p>The <dfn attribute for=AbortController><code>signal</code></dfn> attribute's getter must return
+<a>context object</a>'s <a for=AbortController>signal</a>.
+
+<p>The <dfn method for=AbortController><code>abort()</code></dfn> method, when invoked, must
+<a for=AbortSignal>signal abort</a> on <a>context object</a>'s <a for=AbortController>signal</a>.
+
+
+<h3 id=interface-AbortSignal>Interface {{AbortSignal}}</h3>
+
+<pre class="idl">
+[Exposed=(Window,Worker)]
+interface AbortSignal : EventTarget {
+  readonly attribute boolean aborted;
+
+  attribute EventHandler onabort;
+};</pre>
+
+<dl class=domintro>
+ <dt><code><var>signal</var> . <a attribute for=AbortSignal>aborted</a></code>
+ <dd>Returns true if this {{AbortSignal}} has been aborted, and false otherwise.
+</dl>
+
+<p>An {{AbortSignal}} object has an associated <dfn for=AbortSignal>aborted flag</dfn>. It is unset
+unless specified otherwise.
+
+<p>An {{AbortSignal}} object has associated <dfn for=AbortSignal>abort algorithms</dfn>, which is a
+<a for=/>set</a> of algorithms which are to be executed when its [=AbortSignal/aborted flag=] is
+set. Unless specified otherwise, its value is the empty set.
```

Comment:
I'm not too familiar with the spec language here, but this sounds like it encourages other specs to replace the set, but in reality we only want other things to add/remove item.

Should this say something like "The set is initially empty"?